src/Components/test/E2ETest/Tests/ErrorNotificationServerSideTest.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using BasicTestApp;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.Components.E2ETest.ServerExecutionTests;
+using Microsoft.AspNetCore.E2ETesting;
+using OpenQA.Selenium;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.Tests
+{
+    [Collection("ErrorNotification")] // When the clientside and serverside tests run together it seems to cause failures, possibly due to connection lose on exception.
+    public class ErrorNotificationServerSideTest : ErrorNotificationClientSideTest
+    {
+        public ErrorNotificationServerSideTest(
+            BrowserFixture browserFixture,
+            ToggleExecutionModeServerFixture<Program> serverFixture,
+            ITestOutputHelper output)
+            : base(browserFixture, serverFixture.WithServerExecution(), output)
+        {
+        }
+
+        protected override void InitializeAsyncCore()
+        {
+            // On WebAssembly, page reloads are expensive so skip if possible
+            Navigate(ServerPathBase, noReload: _serverFixture.ExecutionMode == ExecutionMode.Server);
+            Browser.MountTestComponent<ErrorComponent>();
+            Browser.Exists(By.Id("error-ui"));
+            Browser.Exists(By.TagName("button"));
```

Comment:
I don't think this overload of `InitializeAsyncCore` is needed.

The one in the base class is nearly identical, the only change in this derived one is that you put `noReload: _serverFixture.ExecutionMode == ExecutionMode.Server` which looks incorrect to me. I think we only want to skip reloads on ExecutionMode.Client.

So I suggest removing this overload.